pkg/repository/buffer/bulk_events.go

```diff
@@ -79,7 +92,9 @@ func (w *BulkEventWriter) BulkWriteStepRunEvents(ctx context.Context, opts []*re
 	eventData := make([]map[string]interface{}, 0, len(opts))
 	dedupe := make(map[string]bool)
 
-	for i, item := range opts {
+	orderedOpts := sortByStepRunId(opts)
```

Comment:
@abelanger5 any issue with changing the order of the events (keeping them stable wrt to their own step run id)

Reply:
Nope, this looks good 👍